components/sup/src/manager/service/package.rs

```diff
@@ -145,3 +149,57 @@ impl Pkg {
         Ok(pkg)
     }
 }
+
+/// check and see if a user/group is specified in package metadata.
+/// if not, we'll try and use hab/hab.
+/// If hab/hab doesn't exist, try to use (current username, current group).
+/// If that doesn't work, then give up.
+fn get_user_and_group(pkg_install: &PackageInstall) -> Result<(String, String)> {
+    if let Some((user, group)) = get_pkg_user_and_group(&pkg_install)? {
+        Ok((user, group))
+    } else {
+        let defaults = default_user_and_group()?;
+        Ok(defaults)
+    }
```

Comment:
I'm late to the party, but wanted to review for the sake of my `rust` learning. Could this be written as:
```rust
    Ok(get_pkg_user_and_group(&pkg_install)?.unwrap_or(
        default_user_and_group()?,
    ))
```

Reply:
It's hard to know without compiling, because sometimes rustc insists on the creation of temporary variables in order to satisfy its borrow checking logic.  On the surface they seem the same, though to my eye, Matt's code is easier to understand.

Reply:
Oh, I compiled it. I was asking if this was any different logically.

 Good to know your readability preference, though.